test/document_stores/test_document_store.py

```diff
@@ -1391,7 +1391,9 @@ def test_elasticsearch_synonyms():
 @pytest.mark.embedding_dim(384)
 def test_similarity_score(document_store_with_docs):
     retriever = EmbeddingRetriever(
-        document_store=document_store_with_docs, embedding_model="sentence-transformers/paraphrase-MiniLM-L3-v2"
+        document_store=document_store_with_docs,
+        embedding_model="sentence-transformers/paraphrase-MiniLM-L3-v2",
+        model_format="farm",
```

Comment:
I think because of the speed advantage of `_SentenceTransformersEmbeddingEncoder` over `_DefaultEmbeddingEncoder` we will use `_SentenceTransformersEmbeddingEncoder` more often for sentence-transformers models in practice. Our tutorials most often also use `_SentenceTransformersEmbeddingEncoder` explicitly. Therefore, I'd say we should also focus on testing `_SentenceTransformersEmbeddingEncoder` and remove the `model_format` and change the expected scores. It would be best to test both `_DefaultEmbeddingEncoder` and `_SentenceTransformersEmbeddingEncoder`.

Reply:
I am afraid that we currently have no test coverage of `_SentenceTransformersEmbeddingEncoder`. This is something we should definitely change before merging this PR.